lib/SPIRV/libSPIRV/SPIRVFunction.cpp

```diff
@@ -150,7 +150,10 @@ SPIRVFunction::decodeBB(SPIRVDecoder &Decoder) {
 
     SPIRVInstruction *Inst = static_cast<SPIRVInstruction *>(Decoder.getEntry());
     assert(Inst);
-    BB->addInstruction(Inst);
+    if (Inst->getOpCode() == OpUndef)
+      Module->addUndef(Inst->getType());
```

Comment:
addUndef creates Undef instruction with the next available id, which may not match the id of the Undef instruction in the original SPIRV module. Also, what if two function define the same type of Undef?